src/dq_suite/output_transformations.py

```diff
@@ -101,13 +101,50 @@ def round_numeric_params(params: dict) -> dict:
 
 def get_parameters_from_results(result: dict) -> list[dict]:
     """
-    Get the parameters from the GX results.
+    Extracts parameters from a Great Expectations result.
+
+    Behavior:
```

Comment:
I think this function could be rendered much more readable by using clear names for (sub-)functions. 

Now, there is a description of a series of steps in this docstring, that gets repeated in the in-line comments below in slightly different form. None of that would probably be necessary if the various parts got interpretable (sub-)function names. 

The resulting functions could then benefit from comments on 'why' things are happening instead of 'what' is happening (which is clear from the function name).